config/olm/addon-operator-servicemonitor.yaml

```diff
@@ -0,0 +1,26 @@
+apiVersion: monitoring.coreos.com/v1
```

Comment:
Here is an example from the `openshift-apiserver`:

```yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  annotations:
    include.release.openshift.io/ibm-cloud-managed: "true"
    include.release.openshift.io/self-managed-high-availability: "true"
    include.release.openshift.io/single-node-developer: "true"
  name: openshift-apiserver
  namespace: openshift-apiserver
spec:
  endpoints:
    - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
      interval: 30s
      metricRelabelings:
        - action: drop
          regex: etcd_(debugging|disk|server).*
          sourceLabels:
            - __name__
        - action: drop
          regex: apiserver_admission_controller_admission_latencies_seconds_.*
          sourceLabels:
            - __name__
        - action: drop
          regex: apiserver_admission_step_admission_latencies_seconds_.*
          sourceLabels:
            - __name__
        - action: drop
          regex: apiserver_request_duration_seconds_bucket;(0.15|0.25|0.3|0.35|0.4|0.45|0.6|0.7|0.8|0.9|1.25|1.5|1.75|2.5|3|3.5|4.5|6|7|8|9|15|25|30|50)
          sourceLabels:
            - __name__
            - le
      port: https
      relabelings:
        - action: replace
          replacement: openshift-apiserver
          targetLabel: apiserver
      scheme: https
      tlsConfig:
        caFile: /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt
        serverName: api.openshift-apiserver.svc
  namespaceSelector:
    matchNames:
      - openshift-apiserver
  selector: {}
```

Couple of points from my past testing experience on this:
- `tlsConfig.serverName` is required to avoid SAN errors (Subject Alternate Name)
- `spec.endpoints.bearerTokenFile` is also required, can set to the mounted token in the prometheus server
- `tlsConfig.caFile` all services that I've inspected use the already existing ca that is mounted in the prometheus server, this works because the `ServiceMonitor` is owned by the prometheus server scraping the `addon-operator`, so you are really operating from the perspective of that prometheus server, with access to it's files.

Reply:
@sugarraysam for server ca certificate, i followed https://docs.openshift.com/container-platform/4.6/security/certificate_types_descriptions/service-ca-certificates.html to generate and use it. it was given by in-cluster monitoring team

Reply:
but your comments make sense. i will look into it

Reply:
@sugarraysam https://docs.google.com/document/d/1rCKAYTrYMESjJDyJ0KvNap05NNukmNXVVi6MShISnOw/edit and https://github.com/openshift/cluster-etcd-operator/blob/master/manifests/0000_90_etcd-operator_03_servicemonitor.yaml I've updated servicemonitor and removed cm etc. we need prometheus sa,role and rolebinding to enable in-cluster prometheus to discover targets,

Reply:
@sandhyadalavi thanks for the docs. I will have a look. :)